File: Interpretor2/src/Model/OpenRFile.java
package Model;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;

public class OpenRFile implements IStmt {
	
	public String var_file_id;
	public String filename;
	public static Integer key=0;
	
	public OpenRFile(String s1,String s2)
	{
		this.var_file_id=s1;
		this.filename=s2;
		
	}
	public Integer getNextKey()
	{
		return ++this.key;
	}
	public String getFilename()
	{
		return this.filename;
	}
	public String toString()
	{
		return "Open file( "+this.filename+" )";
	}
	@Override
	public PrgState execute(PrgState state)  {
		if(state.fileTable.containsValue(new Tuple<String,BufferedReader>(this.filename,null))==true)
		{
			System.out.println("Filename already exists");
			return null;
		}
		else
		{
			try {
				BufferedReader br=new BufferedReader(new FileReader(this.filename));
				Tuple<String,BufferedReader> t=new Tuple<String,BufferedReader>(this.filename,br);
				Integer i=this.getNextKey();
				state.fileTable.add(i, t);
				state.symTable.add(this.var_file_id, this.key);
			} catch (FileNotFoundException e) {
				e.printStackTrace();
				System.out.println(e);
			}
		}
		return state;
	}

}

File: Interpreter C#/Interpreter/openRFile.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Interpreter
{
    namespace Model
    {
        class openRFile : IStmt
        {
            string var_file;
            string filename;
            int key = 0;

            public openRFile(string v,string f)
            {
                this.var_file = v;
                this.filename = f;
            }

            public int getNextKey()
            {
                return ++this.key;
            }

            PrgState IStmt.execute(PrgState state)
            {
                //IStmt stmt = state.getExeStack().pop();
                if (state.getFileTable().containsValue(new Tuple<String, StreamReader>(this.filename, null)) == 1)
                {
                    Console.WriteLine("Filename already exists");
                    return null;
                }
                else
                {
                    try
                    {
                        StreamReader br = new StreamReader(this.filename);
                        Tuple<String, StreamReader> t = new Tuple<String, StreamReader>(this.filename, br);
                        int i = this.getNextKey();
                        state.getFileTable().add(i, t);
                        state.getSymTable().add(this.var_file, this.key);
                    }
                    catch (FileNotFoundException e)
                    {

                    }




                    return state;

                }
            }

            public override string ToString()
            {
                    return "open File (" + this.filename+") having variable "+this.var_file+"\n";
             }
        }
    }
}
